crates/biome_js_analyze/src/lint/nursery/use_semantic_elements.rs

```diff
@@ -43,7 +43,11 @@ declare_rule! {
 
 impl Rule for UseSemanticElements {
     type Query = Aria<JsxOpeningElement>;
-    type State = AnyJsxAttribute;
+    type State = (
+        AnyJsxAttribute,
+        Option<Vec<String>>,
+        Option<Vec<(String, String)>>,
+    );
```

Comment:
This is a data structure that is very difficult to document (for example, what are third and second arguments?). I strongly advice you to use a `struct` and document the fields. Also, you don't need to use a `Option<Vec>`, you can use directly a `Vec`, and inside the code action you can use `!is_empty()` instead of `is_some`.